### PR TITLE
[Commands] Cleanup #zonelock Command.

### PR DESCRIPTION
**Pull request by Kinglykrab**
- Add support for Zone IDs.
- Cleanup messages and display.
- Fix dangling pointer in ZoneLongName() helper method so name is displayed properly.
image
image

### CODE REVIEW

File: world/zoneserver.cpp [hunk 1]
```diff
@@ -1022,22 +1022,44 @@ void ZoneServer::HandleMessage(uint16 opcode, const EQ::Net::Packet &p) {
...
-				this->SendEmoteMessageRaw(s->adminname, 0, 0, 0, "Failed to change lock");
+		} else if (
+			lock_zone->op == EQ::constants::ServerLockType::Lock ||
+			lock_zone->op == EQ::constants::ServerLockType::Unlock
+		) {
+			if (zoneserver_list.SetLockedZone(lock_zone->zoneID, lock_zone->op == EQ::constants::ServerLockType::Lock)) {
+				zoneserver_list.SendEmoteMessage(
+					0,
+					0,
+					80,
```
Comment (joligario) [inline world/zoneserver.cpp:1038]: Did they ever come up with a common GM status enum? If so, we should switch to it.

**Review by joligario — APPROVED** [review]

File: world/zoneserver.cpp (hunk 1 above)
Comment (Kinglykrab) [reply world/zoneserver.cpp:1038]: If not I can add one.

**Comment by Akkadius** [comment]
Could add some Locked / Unlocked saylinks here while you're at it

**Review by Akkadius — APPROVED** [review]

**Comment by Kinglykrab** [comment]
Adding saylinks to the world stuff requires a change in more than just the include to get them to work and seems kind of out of scope for this, maybe they can be added in the future at some point once that change is made?

**Pull request merged**